### PR TITLE
Added Parameter Objects Headers With Dup Name query for OpenAPI #2909

### LINKED ISSUE #2909: Add Parameter Objects Headers With Dup Name query for OpenAPI
### Platform
OpenAPI 3.0

### Description
*Give a short description about the query*
Parameter Objects should not have duplicate names for 'header' location, since HTTP headers are not case sensitive.

### PR DESCRIPTION
**Pull request by joaoReigota1**
Signed-off-by: João Reigota <joao.reigota@checkmarx.com>

Closes #2909

**Proposed Changes**
- Added Parameter Objects Headers With Dup Name query for OpenAPI 

I submit this contribution under Apache-2.0 license.

### CODE REVIEW

File: assets/queries/openAPI/parameter_objects_headers_dup_name/query.rego [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package Cx
+
+import data.generic.openapi as openapi_lib
+
+CxPolicy[result] {
+	doc := input.document[i]
+	openapi_lib.check_openapi(doc) != "undefined"
+	params := doc.paths[n].parameters
```
Comment (felipe-avelar) [inline assets/queries/openAPI/parameter_objects_headers_dup_name/query.rego:8]: Shouldn't [operations object](https://swagger.io/specification/#operation-object) be checked?

File: assets/queries/openAPI/parameter_objects_headers_dup_name/metadata.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "id": "05505192-ba2c-4a81-9b25-dcdbcc973746",
+  "queryName": "Parameter Objects Headers With Dup Name",
```
Comment (felipe-avelar) [inline assets/queries/openAPI/parameter_objects_headers_dup_name/metadata.json:3]: I think Duplicated would be better than Dup...

**Review by felipe-avelar — DISMISSED** [review]
Please, check comments

File: assets/queries/openAPI/parameter_objects_headers_dup_name/query.rego [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+	check_dup(params)
+
+	result := {
+		"documentId": doc.id,
+		"searchKey": sprintf("openapi.paths.%s.parameters", [n]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "Parameter Object with location 'header', doesn't have duplicate names",
+		"keyActualValue": "Parameter Object with location 'header', has duplicate names",
```
Comment (rafaela-soares) [inline assets/queries/openAPI/parameter_objects_headers_dup_name/query.rego:17]: Maybe the comma could be removed. Same for lines 32 and 33

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Pull request merged**